content/en/docs/application-store/built-in-apps/harbor-app.md

```diff
@@ -6,5 +6,101 @@ description: 'How to use built-in Harbor registry'
 
 weight: 2242
 ---
+From the [Introduction](../../_index) section, you know there was uncounted application could be installed by helm. [kubesphere\'s App Store](https://charts.kubesphere.io/main/) also added some popular application.
 
-TBD
+This tutorial walks you through an example of how to deploy [Harbor](https://goharbor.io/) with several click in kubesphere.
+
+## Prerequisites
+
+- Please make sure you [enable the OpenPitrix system](https://kubesphere.io/docs/pluggable-components/app-store/). We will deploy Harbor from the App Store.
+- You need to create a [workspace, a project, and a user account](https://kubesphere.io/docs/quick-start/create-workspace-and-project/) for this tutorial. The account needs to be a platform regular user and to be invited as the project operator with the `operator` role. In this tutorial, you log in as `project-operator` and work in the project `demo` in the workspace `demo-wp`.
+
+## Hands-on Lab
+
+### Common steps
+
+1. To begin with, you deploy Harbor from the App Store. Please make sure you are landing on the **Overview** page of the project `demo`.
+
+![overview_of_demo_project](/images/docs/appstore/harbor/overview_of_demo_project.png)
+
+2. List your deployed apps. If it is empty, click the button `Deploy New Application`.
+
+![new_app_deploy](/images/docs/appstore/harbor/new_app_deploy.png)
+
+3. Choose harbor template `From App Store`.
+
+![choose_app_from_store](/images/docs/appstore/harbor/choose_app_from_store.png)
+
+4. Find harbor in App Store.
+
+![overview_of_apps](/images/docs/appstore/harbor/overview_of_apps.png)
+
+5. Choose harbor **version** and **deployment location**, then click `Next`.
+
+![deploy_set_of_harbor](/images/docs/appstore/harbor/deploy_set_of_harbor.png)
+
+6. Config harbor yaml, then click `Deploy`. There was an example yaml in section **FAQ**.
+
+![config_of_harbor_deploy](/images/docs/appstore/harbor/config_of_harbor_deploy.png)
+
+{{< notice warning >}}
+Don't forget to edit **externalURL**.
+{{</ notice >}}
+
+7. Check the status of deployment, then try to login harbor.
+
```

Comment:
You are supposed to demonstrate how to access its login page and how to log in to Harbor.

Reply:
The current steps are too simple to follow it.

Reply:
there was three [ways](https://kubernetes.feisky.xyz/concepts/objects/service) they can choose to expose their service.  the example yaml used nodePort.

Reply:
Absolutely I know it, let us introduce one of them to users in this guide.

Reply:
ok